ContextApp/src/main/java/com/github/migi_1/ContextApp/MainActivity.java

```diff
@@ -0,0 +1,194 @@
+        posHolder = PositionHolder.getInstance();
+        
+        // Start the log manager
+        LogManager.getLogManager().getLogger("").setLevel(Level.INFO);
+        
+    }
+    
+    /**
+     * Instanciate the game instance.
+     * Instanciate the sensor manager.
+     * @param savedInstanceState 
+     */
+    @Override  
+    public void onCreate(Bundle savedInstanceState) {  
+        
+        super.onCreate(savedInstanceState);
+
+        //instantiate the application
+        application = (Main) getJmeApplication();
+        application.setDisplayFps(false);
+        application.setDisplayStatView(false);
+
+        //start the sensor manager
+        mSensorManager = (SensorManager) getSystemService(SENSOR_SERVICE);
+        
+        setContentView(R.layout.android_searching);  
+
+        client = com.github.migi_1.ContextApp.client.AutoConnector.getInstance()
+                .autoStart(Executors.newFixedThreadPool(10));
+            
+        // create te accelerometerSensor
+        accelerometerSensor = new AccelerometerSensor(this, client);
+        
+        // wait until position is received
+        /*while (true) {
+            if (posHolder.getPosition() != null) {
+                position = posHolder.getPosition();
+            	break;
+       	    }
+        }*/
+        position = PlatformPosition.FRONTLEFT;
+        
+        setUI();
+        
+    }
+    
+   /**
+    * This method runs the app is resumed.
+    */
+    @Override  
+    protected void onResume() {  
+        super.onResume();
+
+        client.startClient();
+        
+        // register the lister for the accelerometer
+        mSensorManager.registerListener(accelerometerSensor, 
+                mSensorManager.getDefaultSensor(Sensor.TYPE_ACCELEROMETER),
+                SensorManager.SENSOR_DELAY_FASTEST);
+    }
+    
+    /**
+     * Closes the app.
+     */
+    @Override
+    protected void onStop() {  
+        // unregister the sensor listener
+        mSensorManager.unregisterListener(accelerometerSensor);
+            
+        client.closeClient();
+            
+        super.onStop();  
+    } 
+    
+    /**
+     * Sets the UI of the android app in-game, including buttons and images.
+     */
+    public void setUI() {
+        atkMessenger = new AttackMessenger(this);
+        mbFunctions = new MakeButtonFunctions(this);
+        huFunctions = new HeartsUpdateFunctions(this);
+        
+        setContentView(R.layout.android_ingame);
+        
+        TextView textView = (TextView) findViewById(R.id.Location);
+        textView.setText(position.getPosition());
+        
+        mbFunctions.setButtons(position);
+    }
+    
+    /**
+     * Makes sure buttonpresses are logged and processed.
+     * @param butt = the button to which a clicklistener is set
+     * @param str = message to be logged
+     */
+    public void setButtonClick(Button butt, final String str) {
```

Comment:
Butts aren't buttons. This isn't an R rated project.
Also, instead of str, name it something like "message"